### ISSUE TITLE
2.28.x branch regression: Linux joystick support not enabled

### ISSUE THREAD

**Issue by smcv** [libsdl-org/SDL#8488]
After applying commit 221d6ea8 to the 2.28.x branch, `src/joystick/linux/` is not compiled and gamepads are not detected. Reverting that commit fixes the issue.

I think this is because `configure.ac` has this code:

```
        if test x$enable_joystick = xyes; then
          case $ARCH in
            linux)
                if test "x$ac_cv_header_linux_input_h" = xyes; then                            # <---
                    AC_DEFINE(SDL_JOYSTICK_LINUX, 1, [ ])
                    SOURCES="$SOURCES $srcdir/src/joystick/linux/*.c"
                    SOURCES="$SOURCES $srcdir/src/joystick/steam/*.c"
                    have_joystick=yes
                fi
            ;;
```

which assumes that `AC_CHECK_HEADERS([linux/input.h])` has already been done, and checks for a variable that it defines. If `linux/input.h` is dropped from the list of headers to check for, then `$ac_cv_header_linux_input_h` will never be defined.

I don't think the Autotools part of #8459 should have been applied to 2.28.x unless there is a genuine bug that it fixes - if it's just cleanup/refactoring, then please err on the side of not touching stable-branches.

I think it also shouldn't have been applied to development branches without testing that a Linux build with Autotools can still detect joysticks/gamepads.

cc @sezero @madebr @slouken

**Comment by smcv** [comment]
Replacing `test "x$ac_cv_header_linux_input_h" = xyes` with `x$use_input_events = xyes`  (twice: once for Linux and once for FreeBSD) might resolve this.

**Comment by slouken** [comment]
> if it's just cleanup/refactoring, then please err on the side of not touching stable-branches.

Yes, this is our policy. @libsdl-org/a-team, please only merge safe and important changes to stable branches. I'll go ahead and revert that commit.

@sezero, @madebr, can you address the issue in SDL2?

**Comment by sezero** [comment]
I'm working on it

**Comment by sezero** [comment]
Testing this patch now:

```diff
diff --git a/configure.ac b/configure.ac
index 7b71e43..fc2d933 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2755,24 +2755,25 @@ dnl             Work around that we don't have Objective-C support in autoconf
 }
 
 dnl See if we can use the new unified event interface in Linux 2.4
 CheckInputEvents()
 {
 dnl Check for Linux 2.4 unified input event interface support
+        AC_CHECK_HEADERS(linux/input.h)
+
         AC_MSG_CHECKING(for Linux 2.4 unified input interface)
         use_input_events=no
         AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
           #include <linux/input.h>
         ]], [[
           #ifndef EVIOCGNAME
           #error EVIOCGNAME() ioctl not available
           #endif
         ]])], [use_input_events=yes],[])
         AC_MSG_RESULT($use_input_events)
         if test x$use_input_events = xyes; then
-            AC_DEFINE(HAVE_LINUX_INPUT_H, 1, [ ])
             AC_DEFINE(SDL_INPUT_LINUXEV, 1, [ ])
             SUMMARY_input="${SUMMARY_input} linuxev"
         fi
 }
 
 dnl See if we can use the kernel kd.h header
```

**Comment by sezero** [comment]
Fixed in SDL2 branch by 7555701defb2837843d588f50304f7f0f767356d (above patch)

Sam reverted those commits from release-2.28.x, so it should be fine too. Closing.
